torch/csrc/distributed/c10d/NCCLUtils.hpp

```diff
@@ -805,6 +813,77 @@ struct NCCLTraceBuffer {
     }
   }
 
+  std::list<json> getCollectiveTraceJson(bool onlyActive) {
+    auto result = dump_entries();
+
+    std::list<json> entries;
+    for (auto i : c10::irange(result.size())) {
```

Comment:
its a bit error-prone to have 2 methods for parsing 'dump_entries()' output and converting their fields one by one.

i'm not sure what to do instead though.

Reply:
I'll see if I can templatize this in a follow-on review.